branchpro/models.py

```diff
@@ -255,7 +255,164 @@ def simulate(self, parameters, times):
+        """
+        Updates proportionality constant of the R number for imported cases
+        with respect to its analog for local ones.
+
+        Parameters
+        ----------
+        new_epsilon
+            new value of constant of proportionality.
+
+        """
+        self.epsilon = new_epsilon
+
+    def set_imported_cases(self, times, cases):
+        """
+        Sets number of imported cases and when they occur.
+
+        Parameters
+        ----------
+        times
+            times at which imported cases occur. Must be an ordered sequence,
+            without duplicates, and without negative values.
+        cases
+            number of imported cases at that specified point.
+
+        """
+        # Raise error if not correct dimensionality of inputs
+        if np.asarray(times).ndim != 1:
+            raise ValueError(
+                'Times of arising imported cases storage format must \
+                    be 1-dimensional'
+                )
+        if np.asarray(cases).ndim != 1:
+            raise ValueError(
+                'Number of imported cases storage format must be \
+                    1-dimensional')
+
+        # Raise error if inputs do not have same shape
+        if np.asarray(times).shape != np.asarray(cases).shape:
+            raise ValueError('Both inputs should have same number of elements')
+
+        self._imported_times = np.asarray(times)
+        self._imported_cases = np.asarray(cases)
+
+    def simulate(self, parameters, times):
+        """
+        Runs a forward simulation with the given ``parameters`` and returns a
+        time-series with incidence numbers corresponding to the given ``times``
+        .
+
+        Parameters
+        ----------
+        parameters
+            Initial number of cases.
+        times
+            The times at which to evaluate. Must be an ordered sequence,
+            without duplicates, and without negative values.
+            All simulations are started at time 0, regardless of whether this
+            value appears in ``times``.
+
+        """
+        initial_cond = parameters
+        last_time_point = np.max(times)
+
+        # Repeat final r if necessary
+        # (r_1, r_2, ..., r_t)
+        if len(self._r_profile) < last_time_point:
+            missing_days = last_time_point - len(self._r_profile)
+            last_r = self._r_profile[-1]
+            repeated_r = np.full(shape=missing_days, fill_value=last_r)
+            self._r_profile = np.append(self._r_profile, repeated_r)
+
+        incidences = np.empty(shape=last_time_point + 1)
+        incidences[0] = initial_cond
+
+        # Create vector of imported cases
+        imported_incidences = np.zeros(last_time_point + 1, dtype=int)
+
+        imports_times = self._imported_times[
+            self._imported_times <= last_time_point]
+        imports_cases = self._imported_cases[
+            self._imported_times <= last_time_point]
+
+        np.put(
```

Comment:
It seems that np.put wants integers only in ind, causing this line to fail with settings such as imported_times = [1, 2.0]. One fix would be to cast to int when creating self._imported_times (line 360).